### DISCUSSION TITLE
SMBus Documentation of  RDDRONE-BMS772

### DISCUSSION THREAD

**Discussion by MallikarjunSE** [NXP-Robotics/RDDRONE-BMS772#8] [Q&A] [answered]
Hello,
Where can I find SMBus implementation document for the board?
It'd be great if someone can attach the link.
Thank you!

**Comment by cisvanmierlo** [comment, marked as answer]
Hello @MallikarjunSE, 

The documentation can be found in the release notes, see https://github.com/NXPHoverGames/RDDRONE-BMS772/blob/main/BMS772_releaseNotes_4.0.pdf.

For SMBus head over to chapter 7.5 "How to use SMBus".

**Reply by MallikarjunSE** [reply]
Hey @cisvanmierlo 
Can communication happen via UART? I'd prefer UART over smbus. Currently I don't have time for UAVCAN. 
Any details that you could share for UART communication?
Thank you.

**Reply by cisvanmierlo** [reply]
Hi @MallikarjunSE,

There are several ways to communicate with the BMS.

Indeed there are several UAVCAN messages implemented.
Via SMBus  (I2C) the SBS V1.1 messages can be accessed.
Do not forget to enable the SMBus interface with "bms set smbus-enable 1" via the CLI (UART).

Furthermore the command line interface (CLI) can be used to get the data over UART.
With "bms show all 1" or "bms show top 1" it will cyclically send the updated data over UART.

**Reply by MallikarjunSE** [reply]
Thank you! I could get most of the things working on bms. But charging has become issue. After few minutes of charging, I get this message.
![image](https://user-images.githubusercontent.com/49279505/150293243-b16f6b68-c3f6-4e7c-ae1b-92cdfde01d2a.png)
Any suggestions?

**Reply by cisvanmierlo** [reply]
Hi @MallikarjunSE,

It looks like the BMS is measuring an output voltage of 30.9V, which is out of the specified maximum voltage of 26V. (6-cells max). You have to adapt your charger voltage and max current towards the specs of your battery pack. You need to adhere to Constant Current/Constant Voltage (CC/CV) charging protocol. 
The BMS itself is not a charger, but a monitoring system with protections incase voltages and currents are above safe values.

Please note that this is not an user end-product but a development tool with example software, so operate with care.

**Reply by MallikarjunSE** [reply]
Hey @cisvanmierlo Absolutely. We are doing preliminary tests before we start our board design to better understand it.
Thank you.

**Reply by MallikarjunSE** [reply]
> It looks like the BMS is measuring an output voltage of 30.9V, which is out of the specified maximum voltage of 26V. (6-cells max)

Hey @cisvanmierlo But wont there be voltage spikes bigger than 26V in 6S packs during connection?

**Reply by cisvanmierlo** [reply]
Hi @MallikarjunSE,

As I see in the CLI output, there is a constant voltage on the BMS output that is out of the specification.
That is different than short voltage peaks caused by switching off an inductive load.
I would advise you to stay within the specified parameters with the battery- and charger voltage.
